ansible_ai_connect/main/views.py

```diff
@@ -140,6 +140,11 @@ def get_context_data(self, **kwargs):
             context["user_name"] = user.username
         context["debug"] = "true" if settings.CHATBOT_DEBUG_UI else "false"
 
+        llm: ModelPipelineChatBot = apps.get_app_config("ai").get_model_pipeline(
```

Comment:
Hint for optimisation (not that what you have written is slow/expensive to call).

You could have [got the instance](https://github.com/ansible/ansible-ai-connect-service/blob/main/ansible_ai_connect/ai/api/views.py#L967) in a `__init__` function...

... and used this in [L#146](https://github.com/ansible/ansible-ai-connect-service/pull/1516/files#diff-200d781ebcc1aa3378b97b868def96888c9e2084e460c4ead5bcf276d319caa3R146) and [L#126-L#130](https://github.com/ansible/ansible-ai-connect-service/blob/aadf74ea75303fee8811651d547b6ff4ee18b38f/ansible_ai_connect/main/views.py#L126-L130).

Reply:
@manstis I have update the code based on your comment. Pls re-review this when you have time. Thanks!